### ISSUE TITLE
Pull says completed successfully, but nothing is marked as watched

### ISSUE THREAD

**Issue by aptalca** [trakt/Plex-Trakt-Scrobbler#329]
Hi,

Ever since updating to pms 0.9.17.2 trakt no longer syncs the watched data from the trakt server to the plex server. The sync process runs and reports to have been completed successfully, but media is not marked as watched on plex.

I'm using 1.0.3.2-master with pms 0.9.17.2 on ubuntu (docker)

Here's a plugin log: http://pastebin.com/ri9uR94C

Thanks

**Comment by fuzeman** [comment]
Have you tried a "Pull" sync (instead of "Full" or "Quick Pull")? failed syncs can result in some changes to your Trakt.tv profile being missed, running a "Pull" sync will ensure everything matches your Trakt.tv profile.

**Comment by aptalca** [comment]
I tried just the "pull" and get the same result. It says completed successfully but recent items marked as watched on trakt are not marked as watched on plex 

Thanks

**Comment by aptalca** [comment]
Did some more tests. 

I updated the trakt pin (just in case) 

1. I marked a movie as watched in plex (manually) and did a full sync, movie was marked on trakt servers (full sync pushes successfully to trakt) 
2. I watched a show in plex, it was successfully scrobbled to trakt. 
3. I tried a "pull", says it worked, but the couple shows and movies that were watched in kodi and marked as watched on trakt, did not get marked on plex. 
4. I tried a quick pull, same result. 

please let me know if you need more logs with specific actions. 

By the way, the two media items I'm using to test are the tv episode of Orphan Black S04E09 and the movie 10 Cloverfield Lane (in case you need to look for them in the logs). Both are marked as watched on trakt, but not on plex

Thanks

**Comment by hstamas** [comment]
exact same thing happening to me. Trakt will push from Plex successfully but anything that was changed directly on the Trakt website or if a change was initiated by watching something on my Kodi box and then pushed by the Kodi Trakt addon, Plex refuses to reflect the change (pull)

**Comment by aptalca** [comment]
Uninstalled trakt.tv and deleted all related data under Plug-ins Support and reinstalled from UAS2.

The first full sync took 5 min, same result. Nothing new marked as watched on plex.

**Comment by Knuta86** [comment]
I've been having the same problem lately. Episodes watched in Plex gets scrobbled to Trakt, but episodes watched in Kodi and scrobbled to trakt isn't being marked as watched in Plex.

On Plex version 1.0.0.2261 now, but the problem has been present for the last 3 updates at least.

**Comment by Dou9D** [comment]
As per request on plex trakt forum. Please find attached my logs.

I have reinstalled PMS and Trakt plugin and yet it still doesn't function. It used to work fine. I am not sure when it may have stopped working as I have not been watching the same show on my Kodi/Plex setup. Suffice to say, since I have freshly reinstalled PMS my whole collection is marked as unwatched.

Attached a "Full Sync" Debug Log. I have then gone onto the Trakt website, marked Firefly S01E01 as watched and attempted a Pull from Trakt from within PMS. Again nothing is marked as watched on PMS. Kodi has updated successfully

My agents are IMDB for movies (i.e. guid="com.plexapp.agents.imdb://tt0492931?lang=en") and TVBD for tv (i.e. guid="com.plexapp.agents.thetvdb://78874/1/1?lang=en") from the XML info of each plex entry.

Scrobbling an episode DOES work!

[fullsync.zip](https://github.com/trakt/Plex-Trakt-Scrobbler/files/333146/fullsync.zip)
[fireflys01e01pullrequest.zip](https://github.com/trakt/Plex-Trakt-Scrobbler/files/333148/fireflys01e01pullrequest.zip)

**Comment by evulhotdog** [comment]
I saw this a few weeks ago, and then am just now experiencing the same outcome as the others above. I am on:

Plex Ver: 1.0.0.2261
Plugin Ver: 1.0.3.2-master

Tried the same things as above. Full sync does not mark things as watching in Plex, when marked as watched in Trakt. Tried Pull from Trakt.tv and this did also not work as marking it as watched in Plex. I also tried restarting services as well, with no luck. 

Seems to be an identical issue to above, please let me know if you need more information.

**Comment by gdolla** [comment]
Omg please  help
Sent from my Sprint Samsung Galaxy Note5.
-------- Original message --------From: evulhotdog <notifications@github.com> Date: 6/25/16  10:44 AM  (GMT-08:00) To: trakt/Plex-Trakt-Scrobbler <Plex-Trakt-Scrobbler@noreply.github.com> Subject: Re: [trakt/Plex-Trakt-Scrobbler] Pull says completed successfully,
  but nothing is marked as watched (#329) 
I saw this a few weeks ago, and then am just now experiencing the same outcome as the others above. I am on:

Plex Ver: 1.0.0.2261

Plugin Ver: 1.0.3.2-master

Tried the same things as above. Full sync does not mark things as watching in Plex, when marked as watched in Trakt. Tried Pull from Trakt.tv and this did also not work as marking it as watched in Plex. I also tried restarting services as well, with no luck. 

Seems to be an identical issue to above, please let me know if you need more information.

**Comment by TCarneiro91** [comment]
Exactly the same problem that i have on my Plex Media Server...

**Comment by fuzeman** [comment]
Looking at the log files provided by @Dou9D, there appears to be an issue actually updating watched states in Plex as the plugin is attempting to update the same items every sync:
```
2016-06-25 14:57:45,029 - "GET /:/scrobble?identifier=com.plexapp.plugins.library&key=68 HTTP/1.1" 200 0
```
```
2016-06-25 15:10:52,631 - "GET /:/scrobble?identifier=com.plexapp.plugins.library&key=68 HTTP/1.1" 200 0
```

This likely means the plugin is attempting to update watched states, but for some reason the "update" requests are failing...

Also found that the log files provided by @aptalca and @Dou9D both indicate anonymous Plex authentication is being used, so that might be related to this issue.

----

If you haven't already, please post your log files after running a sync (see [this page](https://github.com/trakt/Plex-Trakt-Scrobbler/wiki/Log-files) for details), this will help me identify what is causing this issue.

I'll attempt to reproduce this issue later tonight, will try my best to get a fix released this week :smiley:

**Comment by TCarneiro91** [comment]
Here you have my log, i did a sync before copying this file

https://dl.dropboxusercontent.com/u/26485633/com.plexapp.plugins.trakttv.log

**Comment by Dou9D** [comment]
@fuzeman 
Here is another scenario I have just tested and it was what I was trying to do originally.

Share my PMS server (dougandjenn) with my parents plex account (bkandjad)

- Rules created in the trakt interface ([Screenshot](http://imgur.com/AOHVjbG))
- Filter rules created in PMS trakt settings ([Screenshot](http://imgur.com/k6l3lqW))
- Gotham S01E01 played on plex bkandjad and Firefly S01E01 played on plex dougandjenn ([Screenshot](http://imgur.com/5ldcINQ))
- Gotham scrobbles successfully to the correct trakt account ([Screenshot](http://imgur.com/ZWXz8WP))
- Firefly scrobbles successfully to the correct trakt account ([Screenshot](http://imgur.com/B5xVAp6))

Looks like scrobbling and the routing of scrobbling requests is working (and is really neat!)

- Attempted 'pull from trakt' on account dougandjenn. Fails as Firefly S01E01 still in unwatched state.
- Attempt 'full' sync from bkandjad to upload library. Fails as trakt collection is still empty.

Log file attached.
[trakt_log.zip](https://github.com/trakt/Plex-Trakt-Scrobbler/files/333590/trakt_log.zip)

**Comment by fuzeman** [comment]
Just pushed a fix to the development branch for the anonymous authentication issue, download here: https://github.com/trakt/Plex-Trakt-Scrobbler/archive/develop.zip

Please try a "Pull" sync with this branch installed, and let me know if there is still any syncing issues.

------------

@Dou9D Did you enable collection syncing on the 2nd account via the "Accounts" tab at http://trakt-for-plex.github.io/configuration? syncing needs to be configured per-account.

**Comment by evulhotdog** [comment]
So this seemed to do the trick @fuzeman.

First I tried the full sync, which did not work or update the show that was marked as watched in trakt, but not on Plex. 
I then did a full pull and this worked, and updated the status of that show and several others.

What I'm not understanding is why the Full Sync function is not working. Why would anyone ever use the pull over full sync?

Maybe I'm not understanding the purpose, but in my mind the full sync would be the go to for pretty much everybody.

**Comment by Dou9D** [comment]
All is working again using the developmental branch. I have to say being able to route separate plex logins to separate trakt accounts from a single PMS is pretty swanky! It works really well. 

@fuzeman - *ahem*  I did forget to enable the sync collection in the new Accounts tab. Once I did this the full sync did it's business.

I notice in the web based accounts section it now correctly displays our username rather than 'Adminstrator'.

**Comment by hstamas** [comment]
Dev update seems to have fixed my issues as well. Both a full sync and a individual pull work as expected.

**Comment by aptalca** [comment]
I can confirm that with the develop version, pull completes successfully.

I first tried a full sync, didn't work (I guess it only does a quick pull). I followed that up with a "pull" which worked. I have not tried a full sync again (nothing new watched in kodi yet). Will test it tonight.

Thanks

**Comment by jeffreydeleeuw** [comment]
thanx, could'nt get my head around why it wasnt syncing, happened about the time i installed subzero subtitle plugin, thought that was the problem, but seems fixed with the new develop branch (quick pull, somehow normal pull didnt work)

**Comment by evulhotdog** [comment]
@fuzeman 
I marked 'Fuck, that's delicious' S02E01 as unwatched, ran the full sync, let it complete, and that episode is still marked as watched in Plex. 

Any idea? Log attached :)

[com.plexapp.plugins.trakttv.log](https://github.com/trakt/Plex-Trakt-Scrobbler/files/335948/com.plexapp.plugins.trakttv.txt)

**Comment by fuzeman** [comment]
Great to hear that fix worked, planning on including this fix in an actual release later this week after some more testing.

-----------

@evulhotdog
> What I'm not understanding is why the Full Sync function is not working. Why would anyone ever use the pull over full sync?

"Full" sync (and "Quick Pull") only imports changes from your Trakt.tv profile since the previous sync. This is designed to speed up automated syncs, but can result in missed changes if a sync fails for some reason.

"Pull" sync will compare every item in Plex against your profile, and update anything that is out of date. This should only be needed if "Full" (or "Quick Pull") sync misses a change to your profile, if you need to run this often something is likely broken in the plugin.

> I marked 'Fuck, that's delicious' S02E01 as unwatched, ran the full sync, let it complete, and that episode is still marked as watched in Plex. 

Can't see any obvious issue in that log file, all I can think of is if a "Pull" sync was run after you marked the episode as unwatched on Trakt.tv the change was likely ignored.

**Notes:**
 - You might not of triggered the "Pull" sync yourself, sometimes the Plex/Web interface can trigger extra syncs by itself 😞 
 - Marking items as unwatched in Plex will only ever happen with a "Full" or "Quick Pull" sync, this is due to the difficulties of deciding which state should "win" between Plex and Trakt.tv.

----------------

@Dou9D
> I notice in the web based accounts section it now correctly displays our username rather than 'Adminstrator'.

Yeah, found what was causing this issue as well :smile:

**Comment by fuzeman** [comment]
Fix released in [v1.0.3.3](https://github.com/trakt/Plex-Trakt-Scrobbler/releases/tag/v1.0.3.3)
